pkgs/tools/text/frawk/default.nix

```diff
@@ -0,0 +1,47 @@
+{ lib
+, rustPlatform
+, fetchCrate
+, libxml2
+, ncurses
+, zlib
+, features ? [ "default" ]
+, llvmPackages_12
+}:
+
+let
+  inherit (builtins) elem;
+in
+
+rustPlatform.buildRustPackage rec {
+  pname = "frawk";
+  version = "0.4.6";
+
+  src = fetchCrate {
+    inherit pname version;
+    sha256 = "sha256-yEdfMikMcsQePxQL1+lma95O1x5z1B7aXAEf8apuGaU=";
+  };
+
+  cargoSha256 = "sha256-osi77Fx8jSfIvAIpThgPbnuJVF/Ydr2/+ROHcDG5ZbA=";
+
+  buildInputs = [ libxml2 ncurses zlib ];
+
+  buildNoDefaultFeatures = true;
+  buildFeatures = features;
+
+  preBuild = lib.optionalString (elem "default" features || elem "llvm_backend" features) ''
+    export LLVM_SYS_120_PREFIX=${llvmPackages_12.llvm.dev}
+  '' + lib.optionalString (elem "default" features || elem "unstable" features) ''
+    export RUSTC_BOOTSTRAP=1
+  '';
+
+  # depends on cpu instructions that may not be available on builders
+  doCheck = false;
```

Comment:
Which ones? Can we use the ``stdenv.hostPlatform.avxSupport`` boolean or similar?

Reply:
avx2 and sse2, would using things like `avxSupport` break the cache?

Reply:
I don't think so, we use that in some places already

Reply:
Wouldn't the package have two different nar hashes then, if avxSupport differs on hydra and on the user's machine?

Reply:
yeah, I think so. It would trigger a rebuild.

Reply:
If that means sometimes some people don't get the cache, I would prefer keeping `doCheck = false`

Reply:
I don't see a good solution to have both the tests and the cache